src/main/java/bio/terra/service/dataset/flight/ingest/IngestUtils.java

```diff
@@ -73,6 +78,71 @@ public static DatasetTable getDatasetTable(FlightContext context, Dataset datase
     return optTable.get();
   }
 
+  /**
+   * We want to validate and sanitize the ingest source file We are looking for a url that matches
+   * the following specification: https://{host}/{container}/{blobName}.csv or
+   * https://{host}/{container}/{blobName}.json NOTE: this definition is probably going to be too
+   * strict. There are valid blob urls that do not match this specification and we may be able to
+   * expand this definition with more testing/requirement gathering
+   *
+   * @param URL
+   * @return
+   */
+  public static BlobUrlParts validateBlobAzureBlobFileURL(String URL) {
+    BlobUrlParts blobUrlParts;
+    try {
+      blobUrlParts = BlobUrlParts.parse(URL);
+    } catch (IllegalArgumentException ex) {
+      throw new InvalidBlobURLException("Blob URL parse failed due to malformed URL.", ex);
+    }
+    String scheme = "https";
+    if (!StringUtils.equals(blobUrlParts.getScheme(), scheme)) {
+      throw new InvalidBlobURLException("Ingest source is not a valid blob url: '" + URL + "'");
+    }
+    // Validate host
+    // Expecting: {storageAccount}.blob.core.windows.net
+    // Where storage account meets the following requirements: "Storage account names must be
+    // between 3 and 24 characters in length and may contain numbers and lowercase letters only"
+
+    String host = blobUrlParts.getHost();
+    int separator = StringUtils.indexOf(host, ".");
+    String storageAccountName = StringUtils.substring(host, 0, separator);
+    if (!storageAccountName.matches("^[a-z0-9]{3,24}")) {
+      throw new InvalidBlobURLException("Ingest source is not a valid blob url: '" + URL + "'");
```

Comment:
Similar to above, it would be helpful to know why the url is invalid and/or what the expected format should be